umpost/conversion_driver_esm1p5.py

```diff
@@ -270,6 +271,14 @@ def convert_esm1p5_output_dir(esm1p5_output_dir):
 
     return succeeded, failed
 
+def success_fail_overlap(succeeded, failed):
+    succeeded_inputs = [succeed_path for succeed_path, _ in succeeded]
+    failed_inputs = [fail_path for fail_path, _ in failed]
+
+    overlap = set(succeeded_inputs) & set(failed_inputs)
+
+    return overlap
```

Comment:
We can make use of set comprehensions here:
```Python
def success_fail_overlap(succeeded, failed):
    succeeded_inputs = {succeed_path for succeed_path, _ in succeeded}
    failed_inputs = {fail_path for fail_path, _ in failed}
    return succeeded_inputs - failed_inputs  # rely on set difference operation
```
... then add a test like:
```Python
def test_success_fail_overlap():
    # ensure the shared "fail" item is skipped
    a = ... a dummy successes data structure
    b = ... a dummy failures data structure
    res =  success_fail_overlap(a, b)   
    assert "fail" not in res
```
That should cover the overlap check.

The overlap check could also be called `safe_removal()`, so the calling code becomes:
```Python
for path in safe_removal():
    os.remove(path)
```
Although this loses the ability to report on files appearing in both lists.

Reply:
> Although this loses the ability to report on files appearing in both lists.

This might not be a problem as there is error reporting on the failed files list already...

Reply:
Good idea, and it also makes sense to add the test. I've added this in in the latest commit!

Reply:
And then fixed the typos I added in...